crates/nil/src/main.rs

```diff
@@ -137,7 +147,7 @@ fn main_diagnostics(args: DiagnosticsArgs) {
     match ret {
         Ok(None) => process::exit(0),
         Ok(Some(max_severity)) => {
-            if max_severity > Severity::Warning {
+            if max_severity >= max_allowed_serverity {
```

Comment:
naming nit 1: `s/serverity/severity`

naming nit 2: "max allowed" sounds like this level is allowed. If I'm reading this code correctly, it's *not* allowed. This is hard to name correctly. Perhaps `fail_at_or_above_severity`?

Reply:
thanks for the suggestions ❤️